api/v1alpha1/embedder_types.go

```diff
@@ -32,8 +32,13 @@ type EmbedderSpec struct {
 	// ServiceType indicates the source type of embedding service
 	ServiceType embeddings.EmbeddingType `json:"serviceType,omitempty"`
 
-	// Enpoint defines connection info
+	// Enpoint defines the connection info
```

Comment:
look like we can have 
```go
type Provider struct {
	Enpoint *Endpoint `json:"endpoint,omitempty"`
	Worker *TypedObjectReference `json:"worker,omitempty"`
}

func (p Provider) Provider() Provider {}
func (p Provider) GetWorker() types.NamespacedName {}
```

then just use it in embedder and llm

Reply:
You are right. This is much clear.